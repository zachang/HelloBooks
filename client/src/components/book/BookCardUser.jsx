import React from 'react';
import PropTypes from 'react-proptypes';

const BookCardUser = props => (

  <div className='col l4 s10 m6 cardsm'>
    <div className='card large sticky-action'>
      <div className='card-image waves-effect waves-block waves-light'>
        <img className='activator' src={ (props.book.book_image !== null) ? props.book.book_image : '../../imgs/default.jpg' }/>
      </div>
      <div className='card-content'>
        <span className='card-title activator grey-text text-darken-4'>
          <small>{props.book.book_name}</small>
          <i className='material-icons right'>
            more_vert
          </i>
        </span>
      </div>
      <div className='card-reveal'
        style={{ wordWrap: 'break-word', backgroundColor: 'rgba(0,0,0,1)', color: '#FFF' }}
      >
        <span className='card-title grey-text text-darken-4'>
          <small>{ props.book.book_name }</small>
          <i className='material-icons right'>
            close
          </i>
        </span>
        <p>Author: { props.book.author }</p>
        <p>Category: { props.book.Category.category_name }</p>
        <p>Published: { new Date(props.book.publish_year).getFullYear() }</p>
        <p>Pages: { props.book.pages }</p>
        <p>Stock: { props.book.book_count }</p>
        <p>Description: {
          (props.book.description)
        }
        </p>
      </div>
      <div className='card-action home-card'>
        <button
          onClick={ () => (props.borrowBook(window.sessionStorage.token,  props.book.id )) }
          className='waves-effect waves-light btn teal'
        >
          Borrow
        </button>
      </div>
    </div>
  </div>
);

BookCardUser.propTypes = {
  book: PropTypes.object.isRequired,

};
export default BookCardUser;
