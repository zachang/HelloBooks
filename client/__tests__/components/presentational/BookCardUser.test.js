import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import BookCardUser from '../../../src/components/book/BookCardUser.jsx';

describe('<BookCardUser/>', () => {
  let wrapper;

  const props = {
    book: {
      bookImage: 'ggv.jpg',
      id: 1,
      bookName: 'Harry Potter',
      publishYear: new Date(),
      bookCount: 5,
      description: 'nice read',
      pages: 700,
      Category: {
        categoryName: 'Arts'
      }
    },
    borrowBook: jest.fn(),
    readBook: jest.fn()
  };

  const wrapperMount = mount(<BookCardUser {...props} />);
  const shallowWrapper = shallow(<BookCardUser {...props}/>);
  const tree = toJson(shallowWrapper);


  it('renders <BookCardUser/> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick', () => {
    wrapper = shallow(<BookCardUser {...props} />);
    wrapper.find('.readBook').simulate('click');
    expect(wrapperMount.find('.readBook').at(0).text()).toBe('remove_red_eye');
  });

  it('calls onClick', () => {
    wrapper = shallow(<BookCardUser {...props} />);
    wrapper.find('.borrowBook').simulate('click');
    expect(wrapperMount.find('.borrowBook').at(0).text()).toBe('dlibrary_books');
  });
});

