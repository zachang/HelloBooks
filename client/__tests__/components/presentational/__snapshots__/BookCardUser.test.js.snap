// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookCardUser/> renders <BookCardUser/> component 1`] = `
<div
  className="col l4 s10 m6 cardsm"
>
  <div
    className="card large sticky-action"
  >
    <div
      className="card-image waves-effect waves-block waves-light"
    >
      <img
        className="activator"
        src="ggv.jpg"
      />
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title activator grey-text text-darken-4"
      >
        <small>
          Harry Potter
        </small>
        <i
          className="material-icons right"
        >
          more_vert
        </i>
      </span>
    </div>
    <div
      className="card-reveal"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,1)",
          "color": "#FFF",
          "wordWrap": "break-word",
        }
      }
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        <small>
          Harry Potter
        </small>
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
      <p>
        Author: 
      </p>
      <p>
        Category: 
        Arts
      </p>
      <p>
        Published: 
        2018
      </p>
      <p>
        Pages: 
        700
      </p>
      <p>
        Stock: 
        5
      </p>
      <p>
        Description: 
        nice read
      </p>
    </div>
    <div
      className="card-action home-card"
    >
      <Link
        className="white tooltipped"
        data-delay="50"
        data-position="bottom"
        data-tooltip="Read book"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="read/1"
      >
        <i
          className="small material-icons readBook"
          onClick={[Function]}
          style={
            Object {
              "color": "teal",
              "cursor": "pointer",
            }
          }
        >
          remove_red_eye
        </i>
      </Link>
      <Link
        className="white tooltipped"
        data-delay="50"
        data-position="bottom"
        data-tooltip="Borrow book"
        onlyActiveOnIndex={false}
        style={Object {}}
      >
        <i
          className="small material-icons borrowBook"
          onClick={[Function]}
          style={
            Object {
              "color": "teal",
              "cursor": "pointer",
            }
          }
        >
          dlibrary_books
        </i>
      </Link>
    </div>
  </div>
</div>
`;
